File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalPaymentIntent.java
package com.braintreepayments.api;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * The payment intent in the PayPal Checkout flow
 */
@Retention(RetentionPolicy.SOURCE)
@StringDef({PayPalPaymentIntent.ORDER, PayPalPaymentIntent.SALE, PayPalPaymentIntent.AUTHORIZE})
public @interface PayPalPaymentIntent {

    /**
     * Payment intent to create an order
     */
    String ORDER = "order";

    /**
     * Payment intent for immediate payment
     */
    String SALE = "sale";

    /**
     * Payment intent to authorize a payment for capture later
     */
    String AUTHORIZE = "authorize";
}

File: PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalPaymentIntent.kt
package com.braintreepayments.api.paypal

import androidx.annotation.RestrictTo

/**
 * The payment intent in the PayPal Checkout flow
 */
enum class PayPalPaymentIntent(internal val stringValue: String) {

    /**
     * Payment intent to create an order
     */
    ORDER("order"),

    /**
     * Payment intent for immediate payment
     */
    SALE("sale"),

    /**
     * Payment intent to authorize a payment for capture later
     */
    AUTHORIZE("authorize");

    companion object {
        @JvmStatic
        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
        fun fromString(string: String?): PayPalPaymentIntent? {
            return entries.firstOrNull { it.stringValue.equals(string, ignoreCase = true) }
        }
    }
}
